credoai/modules/dataset_modules/dataset_base.py

```diff
@@ -156,6 +159,44 @@ def _calculate_mutual_information(self, categorical_features=None, normalize=Tru
 
         return results
     
+    def _assess_balance_metrics(self):
+        """Calculate dataset balance statistics and metrics 
 
-    
+        Returns
+        -------
+        dict
+            'sample_balance': distribution of samples across groups
+            'label_balance': distribution of labels across groups
+            'metrics': maximum statistical parity and maximum disparate impact between groups for all preferred label value possibilities 
+        """
+        credo_data = cl.CredoData(name='dataset',
+                          X=self.X,
+                          y=self.y,
+                          sensitive_features=self.sensitive_features)
+
+        df, sensitive_feature_name, label_name = credo_data._concat_features_label_to_dataframe()
+        results = {}
+
+        # Distribution of samples across groups
+        sb = df.groupby([sensitive_feature_name]).agg(count=(label_name, len), percentage=(label_name, lambda x: 100.0*len(x)/len(df))).reset_index().to_dict(orient='records')
```

Comment:
nit: make multiline

Reply:
nit: `sb` is not immediately understandable

Reply:
applied